internal/api/provider/slack_legacy.go

```diff
@@ -0,0 +1,42 @@
+package provider
+
+import (
+	"strings"
+
+	"github.com/supabase/auth/internal/conf"
+	"golang.org/x/oauth2"
+)
+
+// NewSlackProvider creates a Slack account provider.
+func NewSlackLegacyProvider(ext conf.OAuthProviderConfiguration, scopes string) (OAuthProvider, error) {
```

Comment:
wouldn't this break existing slack legacy oauth apps since the `OAuthProvider` [interface](https://github.com/supabase/auth/blob/e3ebffb10662d4b369478b7da8235ef127e1f7d4/internal/api/provider/provider.go#L88-L92) isn't clearly defined?